ecs-cli/modules/cli/local/project/project.go

```diff
@@ -0,0 +1,214 @@
+	LocalInFileName         = "./task-definition.json"
+)
+
+type LocalProject interface {
+	ReadTaskDefinition() error
+	Convert() error
+	Write() error
+	LocalOutFileName() string
+}
+
+type localProject struct {
+	context          *cli.Context
+	taskDefinition   *ecs.TaskDefinition
+	localBytes       []byte
+	localOutFileName string
+}
+
+func New(context *cli.Context) LocalProject {
+	p := &localProject{context: context}
+	return p
+}
+
+func (p *localProject) TaskDefinition() *ecs.TaskDefinition {
+	return p.taskDefinition
+}
+
+func (p *localProject) LocalOutFileName() string {
+	return p.localOutFileName
+}
+
+func (p *localProject) ReadTaskDefinition() error {
+	arn := p.context.String(flags.TaskDefinitionArnFlag)
+	filename := p.context.String(flags.TaskDefinitionFileFlag)
+
+	if arn != "" && filename != "" {
+		return fmt.Errorf("Cannot specify both --%s and --%s flags.", flags.TaskDefinitionArnFlag, flags.TaskDefinitionFileFlag)
+	}
+
+	var taskDefinition *ecs.TaskDefinition
+	var err error
+
+	if arn != "" {
+		taskDefinition, err = p.readTaskDefinitionFromArn(arn)
+		if err != nil {
+			return err
+		}
+	} else if filename != "" {
+		taskDefinition, err = p.readTaskDefinitionFromFile(filename)
+		if err != nil {
+			return err
+		}
+	} else if _, err := os.Stat(LocalInFileName); err == nil {
+	// Try reading local task-definition.json file by default
+		taskDefinition, err = p.readTaskDefinitionFromFile(LocalInFileName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if taskDefinition == nil {
+		return fmt.Errorf("Could not detect valid Task Definition")
+	}
+
+	p.taskDefinition = taskDefinition
+	return nil
+}
+
+func (p *localProject) readTaskDefinitionFromFile(filename string) (*ecs.TaskDefinition, error) {
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading task definition from %s: %s", filename, err.Error())
+	}
+
+	taskDefinition := ecs.TaskDefinition{}
+	err = json.Unmarshal(bytes, &taskDefinition)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing task definition JSON: %s", err.Error())
+	}
+
+	return &taskDefinition, nil
+}
+
+var newCommandConfig = func(context *cli.Context, rdwr config.ReadWriter) (*config.CommandConfig, error) {
+	return config.NewCommandConfig(context, rdwr)
+}
+
+// FIXME: NOTE this will actually read from either ARN or Task Definition family name
+func (p *localProject) readTaskDefinitionFromArn(arn string) (*ecs.TaskDefinition, error) {
+	rdwr, err := config.NewReadWriter()
+	if err != nil {
+		return nil, err
+	}
+
+	commandConfig, err := newCommandConfig(p.context, rdwr)
+	if err != nil {
+		return nil, err
+	}
+
+	ecsClient := ecsclient.NewECSClient(commandConfig)
+
+	return ecsClient.DescribeTaskDefinition(arn)
+}
+
+func (p *localProject) Convert() error {
+	// FIXME get secrets here, pass to converter?
+	data, err := converter.ConvertToDockerCompose(p.taskDefinition)
```

Comment:
Is there a way for us to add additional Labels to the Compose file?

More specifically `local ps` and `local down` need the following labels: https://github.com/aws/amazon-ecs-cli/pull/784/files#diff-1411f25c19c71b45e043e0d3c3b0ef2eR36

Reply:
There should be -- the converter calls `convertDockerLabels`, so we can add anything needed there.